housekeeper/cli/tables.py

```diff
@@ -87,3 +89,67 @@ def get_versions_table(rows: List[dict]) -> Table:
             version_obj["expires_at"].split("T")[0] if version_obj["expires_at"] else "",
         )
     return table
+
+
+def squash_names(list_of_files):
+    """If subsequent elements (filenames) in 'list_of_files' end in an integer- And that integer is
+    following the previous- those are squashed when displayed.
+    Example:
+
+        ["asdf1.txt", "asdf2.txt"] becomes asdf[1-2].txt'
+    """
+    list_of_squashed = []
+    tag_list = []
+    if list_of_files == []:
+        return list_of_squashed
+    head, *tail = list_of_files
+    (previous_file, previous_counter, previous_suffix) = _get_suffix(head["path"])
+    squash = [previous_counter]
+    previous_hkjson = head
+    for hk_json in tail+[{'path':""}]:  # Add padding for final iteration
```

Comment:
I find it a bit tricky to follow.. What is even head and tail? What are the data types and what do they represent?

Reply:
> I find it a bit tricky to follow.. What is even head and tail? What are the data types and what do they represent?

The _head_ and _tail_ is a convention for representing first element of a list –and the rest. And more readable than the other convention of using _x_ and _xs_. 😀